commit 34f3baa43d49673fe0fb23f05aef4790e2022392
Author: Kai Vehmanen <kvcontact@nosignal.fi>
Date:   Thu Jan 30 22:41:41 2014 +0200

    eca-control-objects: properly init chainsetup edit objs
    
    Always set the value of 'need_chain_reinit' when creating chainsetup
    edit objects. Depending on stack contents, lack of initialization led
    to unnecessary reinitialization of chains for many commands, including
    cop-set/ctrlp-set/cop-bypass/c-mute/c-bypass.

diff --git a/libecasound/eca-control-objects.cpp b/libecasound/eca-control-objects.cpp
index b042da3..6aca2b5 100644
--- a/libecasound/eca-control-objects.cpp
+++ b/libecasound/eca-control-objects.cpp
@@ -1,6 +1,6 @@
 // ------------------------------------------------------------------------
 // eca-control-objects.cpp: Class for configuring libecasound objects
-// Copyright (C) 2000-2004,2006,2008,2009,2012,2013 Kai Vehmanen
+// Copyright (C) 2000-2004,2006,2008,2009,2012-2014 Kai Vehmanen
 // Copyright (C) 2005 Stuart Allie
 //
 // Attributes:
@@ -2027,6 +2027,7 @@ void ECA_CONTROL::set_chain_operator_parameter(CHAIN_OPERATOR::parameter_t value
   ECA::chainsetup_edit_t edit;
   edit.type = ECA::edit_cop_set_param;
   edit.cs_ptr = selected_chainsetup_repp;
+  edit.need_chain_reinit = false;
 
   unsigned int p = selected_chainsetup_repp->first_selected_chain();
   if (p < selected_chainsetup_repp->chains.size()) {
@@ -2074,6 +2075,7 @@ void ECA_CONTROL::set_chain_muting(const string &arg)
   ECA::chainsetup_edit_t edit;
   edit.type = ECA::edit_c_muting;
   edit.cs_ptr = selected_chainsetup_repp;
+  edit.need_chain_reinit = false;
 
   int state_arg = priv_onofftoggle_to_int(arg);
 
@@ -2102,6 +2104,7 @@ void ECA_CONTROL::set_chain_bypass(const string& arg)
   ECA::chainsetup_edit_t edit;
   edit.type = ECA::edit_c_bypass;
   edit.cs_ptr = selected_chainsetup_repp;
+  edit.need_chain_reinit = false;
 
   int state_arg = priv_onofftoggle_to_int(arg);
 
@@ -2132,6 +2135,7 @@ void ECA_CONTROL::bypass_chain_operator(const string& arg)
   ECA::chainsetup_edit_t edit;
   edit.type = ECA::edit_cop_bypass;
   edit.cs_ptr = selected_chainsetup_repp;
+  edit.need_chain_reinit = false;
 
   int bypass_arg = priv_onofftoggle_to_int(arg);
 
@@ -2517,6 +2521,7 @@ void ECA_CONTROL::set_controller_parameter(CHAIN_OPERATOR::parameter_t value)
   ECA::chainsetup_edit_t edit;
   edit.type = ECA::edit_ctrl_set_param;
   edit.cs_ptr = selected_chainsetup_repp;
+  edit.need_chain_reinit = false;
 
   unsigned int p = selected_chainsetup_repp->first_selected_chain();
   if (p < selected_chainsetup_repp->chains.size()) {
